ComfortZoneApp/app/src/main/java/com/example/comfortzone/Utils/LocationUtil.java

```diff
@@ -0,0 +1,96 @@
+package com.example.comfortzone.Utils;
+
+import android.Manifest;
+import android.annotation.SuppressLint;
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.location.Location;
+import android.location.LocationManager;
+import android.os.Looper;
+import android.provider.Settings;
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.core.app.ActivityCompat;
+
+import com.google.android.gms.location.FusedLocationProviderClient;
+import com.google.android.gms.location.LocationCallback;
+import com.google.android.gms.location.LocationRequest;
+import com.google.android.gms.location.LocationResult;
+import com.google.android.gms.location.LocationServices;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+
+public class LocationUtil {
+
+    public static final String TAG = "locationutil";
+    public static final int PERMISSION_ID = 44;
+    public static String lat;
+    public static String lon;
```

Comment:
If you add a callback for getting location, we won't need these. In general, it's bad practice to use static variable to store states, as memory is allocated as long as the app is running. If you need to store a state, you should use member variable, so that if an activity that is using it is dismissed, the object will get deallocated as well.

There are 2 main changes we need for this location util:
1. add callback for `getLastLocation` so that the caller can get updated when there's a location
2. set certain threshold for last location, i.e. if it's older than an hour, then we request a new location to pass back to callback